tools/git-contributors/git-contributors.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python3
+
+# Descent 3
+# Copyright (C) 2024 Descent Developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+#* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from git import Repo
+import re
+
+
+def strip_comments(code):
+    code = str(code)
+    return re.sub(r'(#.*)?\n?', '', code)
+
+
+def main():
+    noadds = list()
+    name_maps = dict()
+    with open("no-add.txt") as noadd_file:
+        for line in noadd_file:
+            name = strip_comments(line)
+            noadds.append(name.strip())
+    with open("map-names.txt") as map_file:
+        for line in map_file:
+            (git_name, replace_name) = strip_comments(line.strip()).split(":")
+            name_maps.update({git_name: replace_name})
+
+    repo = Repo("../..")
+    commits = list(repo.iter_commits("main"))
+    authors = set()
+
+    for commit in commits:
+        if commit.author.name not in noadds:
+            if commit.author.name in name_maps.keys():
+                authors.add(name_maps[commit.author.name])
+            else:
+                authors.add(commit.author.name)
+
+    with open("../../scripts/data/fullhog/oscredits.txt", "w") as result_file:
```

Comment:
please make it a  `argv` variable, so we can easily relocate the script or the output file